lib/src/widget_selector.dart

```diff
@@ -1,170 +1,99 @@
+  }
+
+  @override
+  Widget errorBuild(BuildContext context, Object error) {
+    if (errorBuilder != null) return errorBuilder(context, error);
+    return onFalse;
   }
-  
 }
 
 /// `WidgetBuilderSelector`is a convenience class that will one of two builder methods based on the output of a `Stream<bool>`
 /// This is pretty handy if you want to react to state change like enable/disable in you ViewModel and update
 /// the View accordingly.
-/// In comparrison to `WidgetSelector` this is best used if the alternative child widgets are large so that you don't want to have them created 
-/// without using them. 
-class WidgetBuilderSelector  extends StatelessWidget{
-
-  final Stream<bool> buildEvents;
+/// In comparrison to `WidgetSelector` this is best used if the alternative child widgets are large so that you don't want to have them created
+/// without using them.
+class WidgetBuilderSelector extends ReactiveWidget<bool> {
   final WidgetBuilder onTrue;
   final WidgetBuilder onFalse;
+  final ErrorBuilder<String> errorBuilder;
+  final EmptyBuilder streamEmptyBuilder;
 
   /// Creates a new WidgetBuilderSelector instance
-  /// `buildEvents` : `Stream<bool>`that signals that the this Widget should be updated
+  /// `stream` : `Stream<bool>`that signals that the this Widget should be updated
   /// `onTrue` : builder that should be executed if an item with value true is received
-  /// `onFalse`: builder that should be executed if an item with value true is received
-  const WidgetBuilderSelector({this.buildEvents,  this.onTrue,  this.onFalse, Key key }) 
-          :  assert(buildEvents != null),assert(onTrue != null), assert(onFalse != null), super(key: key);
+  /// `onFalse`: builder that should be executed if an item with value false is received
+  const WidgetBuilderSelector({
+    Stream<bool> stream,
+    this.onTrue,
+    this.onFalse,
+    this.errorBuilder,
+    this.streamEmptyBuilder,
+    Key key,
+    bool initialValue,
+  })  : assert(stream != null),
+        assert(onTrue != null),
+        assert(onFalse != null),
+        super(stream, initialValue, key: key);
 
   @override
-  Widget build(BuildContext context) {
-    return new StreamBuilder(stream: buildEvents,
-                          builder: (BuildContext context, AsyncSnapshot<bool> event)
-                          {   
-                              if (event.hasData && event.data == true)
-                              {
-                                  return onTrue(context);
-                              }
-                              else
-                              {
-                                return onFalse(context);
-                              }
-                          }); 
+  Widget build(BuildContext context, data) {
+    if (data) return onTrue(context);
+    return onFalse(context);
   }
-  
-}
-
-
-/// Spinner/Busyindicator that reacts on the output of a `Stream<CommandResult<T>>`. It's made especially to work together with

```

Comment:
Why did you remove the RxCommandBuilder? I'm pretty sure there a users using it

Reply:
I just moved to the rx_command_builder.dart file.

Reply:
ah, I'm just wondering why the new file isn't listed when doing a review